src/Gui/CommandView.cpp

```diff
@@ -606,18 +625,30 @@ Action * StdCmdToggleClipPlane::createAction()
     return pcAction;
 }
 
-void StdCmdToggleClipPlane::activated(int iMsg)
+void StdCmdToggleClipPlane::updateView(const MDIView *view)
 {
-    Q_UNUSED(iMsg);
-    static QPointer<Gui::Dialog::Clipping> clipping = nullptr;
-    if (!clipping) {
-        auto view = qobject_cast<View3DInventor*>(getMainWindow()->activeWindow());
-        if (view) {
+    const auto view3d = dynamic_cast<Gui::View3DInventor const*>(view);
+    updateView(view3d);
+}
+
+void StdCmdToggleClipPlane::updateView(Gui::View3DInventor const* view)

```

Comment:
This suffers in that, once you open the clipping plane for a view, it re-opens when returning to that view, regardless if the dialog was closed prior or not.